packages/protvista-vcf-adapter/src/protvista-vcf-adapter.ts

```diff
@@ -0,0 +1,90 @@
+import { html, render as litRender } from "lit-html";
+import ProtvistaFeatureAdapter from "protvista-feature-adapter";
+import { NightingaleElement } from "data-loader";
+import { vcfToJSON } from "vcftojson";
+
+import transformData from "./dataTransformer";
+
+class VCFAdapter extends ProtvistaFeatureAdapter implements NightingaleElement {
+  private accession: string;
+
+  private sequence: string;
+
+  constructor() {
+    super();
+    this.attachShadow({ mode: "open" });
+    this.accession = this.getAttribute("accession");
+    this.onChange = this.onChange.bind(this);
+  }
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    if (this.closest("protvista-manager")) {
+      this.manager = this.closest("protvista-manager");
+      this.manager.register(this);
+    }
+    this.render();
+  }
+
+  disconnectedCallback(): void {
+    if (this.manager) {
+      this.manager.unregister(this);
+    }
+  }
+
+  _emitEvent(data?: { sequence: string }): void {
+    if (data) {
+      this.sequence = data.sequence;
+    }
+    // TODO handle lack of sequence better here, could be race condition
+    if (this.sequence && this._adaptedData && this._adaptedData.length) {
+      this.dispatchEvent(
+        new CustomEvent("load", {
+          detail: {
+            payload: {
+              sequence: this.sequence,
+              variants: this._adaptedData,
+            },
+          },
+          bubbles: true,
+          cancelable: true,
+        })
+      );
+    }
+  }
+
+  async parseEntry(data: string): Promise<void> {
+    const vcfJson = await vcfToJSON(data, { runVEP: true });
+    this._adaptedData = transformData(vcfJson, this.accession);
+    this._emitEvent();
+  }
+
+  static get is(): string {
+    return "vcf-adapter";
+  }
+
+  onChange(fileSelect: Event): void {
+    const target = (fileSelect.target as unknown) as HTMLInputElement;
```

Comment:
Maybe doing an early escape with `if (!(fileSelect.target instanceof HTMLInputElement)) return;` would either set the type of target as HTMLInputElement, or at least let you cast it without resorting to `...as unknown as...`